### PR TITLE
introduce new project structure (first step)

### PR DESCRIPTION
**Pull request by flexsurfer**
introduce new project structure (first step)

```
src
├── i18n
├── status_im2
│   ├── common
│   ├── contexts/
│   ├── setup/
│   ├── navigation/
│   └── subs/
└── utils.cljs
```

you can ignore changes in subs and status-im folders

### CODE REVIEW

File: src/hot_reload.cljs [hunk 1]
```diff
@@ -1,12 +1,9 @@
-(ns status-im.reloader
+(ns hot-reload
```
Comment (ilmotta) [inline src/hot_reload.cljs]: I think this namespace could live inside the `src/status-im/setup/` directory. From the docs:

> Contains namespaces that are mostly used to initialize the application, configure test runners, etc. In general, such namespaces should not be required from the outside.

 Although this namespace is required in some places, I see it as boring plumbing not worthy of living at the top level directory.

**Review by ilmotta — APPROVED** [review]

**Comment by OmarBasem** [comment]
I don't know if this is just me, but I do not like numbers in file names and namespaces like `status_im2` `ui2` `quo2`. Why don't we call it something less verbose without numbers? Maybe simply `status` or `app`.

**Comment by flexsurfer** [comment]
> I don't know if this is just me, but I do not like numbers in file names and namespaces like `status_im2` `ui2` `quo2`. Why don't we call it something less verbose without numbers? Maybe simply `status` or `app`.

it's not only you, i don't like as well, it's temporary :)

**Comment by OmarBasem** [comment]
Since this is a major migration, how can we make sure that no functionality gets broken when migrating from `status-im` to `status-im2`? (as some functionalities got broken in the previous migration from `ui` to `ui2`)

**Comment by J-Son89** [comment]
> Since this is a major migration, how can we make sure that no functionality gets broken when migrating from `status-im` to `status-im2`? (as some functionalities got broken in the previous migration from `ui` to `ui2`)

that's why we need more tests :)

**Comment by flexsurfer** [comment]
> Since this is a major migration, how can we make sure that no functionality gets broken when migrating from `status-im` to `status-im2`? (as some functionalities got broken in the previous migration from `ui` to `ui2`)

since we don't have much, and we don't have a release, we're fine to break things, don't worry, we can fix them quickly

File: src/react_native/config.cljs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+(ns react-native.config
+  (:require ["react-native-config" :default react-native-config]))
+
+(def config (js->clj react-native-config :keywordize-keys true))
+
+(defn get-config
+  ([k] (get config k))
+  ([k not-found] (get config k not-found)))
```
Comment (ibrkhalil) [inline src/react_native/config.cljs]: Do we need a new line at EOF ?

File: src/react_native/background_timer.cljs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  (.setTimeout background-timer cb ms))
+
+(defn clear-timeout [id]
+  (.clearTimeout background-timer id))
+
+(defn set-interval [cb ms]
+  (.setInterval background-timer cb ms))
+
+(defn clear-interval [id]
+  (.clearInterval background-timer id))
```
Comment (ibrkhalil) [inline src/react_native/background_timer.cljs]: New line?

File: src/i18n/i18n.cljs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+(defn label-pluralize [count path & options]
+  (if (exists? (.t i18n))
+    (.p i18n count (name path) (clj->js options))
+    (name path)))
+
+(def locale
+  (.-locale i18n))
+
+(def format-currency goog.i18n/format-currency)
```
Comment (ibrkhalil) [inline src/i18n/i18n.cljs]: New line ?

File: src/react_native/core.cljs [hunk 5]
```diff
@@ -28,4 +30,21 @@
...
+(def appearance ^js (.-Appearance ^js react-native))
+
+(defn get-color-scheme []
+  (.getColorScheme appearance))
+
+(defn appearance-add-change-listener [handler]
+  (.addChangeListener appearance handler))
+
+(defn get-window []
+  (js->clj (.get (.-Dimensions ^js react-native) "window") :keywordize-keys true))
```
Comment (ibrkhalil) [inline src/react_native/core.cljs]: New line?

File: src/react_native/languages.cljs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+(ns react-native.languages
+  (:require ["react-native-languages" :default react-native-languages]))
+
+(defn add-change-listener [handler]
+  (.addEventListener ^js react-native-languages "change" (fn [^js event] (handler (.-language event)))))
+
+(defn get-lang-keyword []
+  (keyword (.-language ^js react-native-languages)))
```
Comment (ibrkhalil) [inline src/react_native/languages.cljs]: New line ?

**Review by ibrkhalil — APPROVED** [review]
lgtm
Non blocking comments
Kindly check the newline at end of files

**Comment by OmarBasem** [comment]
> > Since this is a major migration, how can we make sure that no functionality gets broken when migrating from `status-im` to `status-im2`? (as some functionalities got broken in the previous migration from `ui` to `ui2`)
> 
> 
> 
> that's why we need more tests :) 

Right :)

**Review by OmarBasem — APPROVED** [review]

**Comment by OmarBasem** [comment]
@flexsurfer I was trying to work on the [input issue](https://github.com/status-im/status-mobile/issues/14362), but currently there are a few broken things.  Currently, the chat screen is showing old-ui.  Also, no messages list, groups list or contacts list are being displayed on the home screen.

**Comment by flexsurfer** [comment]
> @flexsurfer I was trying to work on the [input issue](https://github.com/status-im/status-mobile/issues/14362), but currently there are a few broken things. Currently, the chat screen is showing old-ui. Also, no messages list, groups list or contacts list are being displayed on the home screen.

@OmarBasem sorry, should be fixed in latest develop

**Pull request merged**